### PR TITLE
Feature/safe logging init use

### PR DESCRIPTION
**Pull request by hsaraogi**
**Goals (and why)**: Test our use of the annotations in safe-logging is legit

**Implementation Description (bullets)**:
1. Add dependency on safe logging and annotations
2. We should look at these log lines in the internal logging product.

**Concerns (what feedback would you like?)**: The UnsafeArg logs dont seem to be on common code paths. This will make testing log collection difficult, Not sure if we have some?

**Where should we start reviewing?**: N/A

**Priority (whenever / two weeks / yesterday)**: one week

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1931)

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java [hunk 1]
```diff
@@ -462,7 +464,7 @@ public String toString() {
     @Override
     public Map<Cell, Value> get(TableReference tableRef, Map<Cell, Long> timestampByCell) {
         if (timestampByCell.isEmpty()) {
-            log.info("Attempted get on '{}' table with empty cells", tableRef);
+            log.info("Attempted get on '{}' table with empty cells", UnsafeArg.of("tableRef", tableRef));
```
Comment (tpetracca) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java]: Why are we considering table names unsafe?

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java [hunk 2]
```diff
@@ -2272,10 +2274,11 @@ public void cleanUpSchemaMutationLockTablesState() throws Exception {
         tables.remove(tableToKeep.get());
         if (tables.size() > 0) {
             dropTablesInternal(tables);
-            log.info("Dropped tables [{}]", tables.toString());
+            log.info("Dropped tables [{}]", UnsafeArg.of("table names", tables.toString()));
```
Comment (schlosna) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java]: Do we really need to wrap unsafe args? There's a non-zero allocation & GC cost to wrapping for logging in tight loops or on hot code paths (though this specific case is not one).

Also realize this was like this before, but we should not be calling `toString()` on any log args as that kind of defeats the point of lazy evaluation if we're not logging at that log level.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java [hunk 3]
```diff
@@ -121,6 +121,8 @@
 import com.palantir.common.base.FunctionCheckedException;
 import com.palantir.common.base.Throwables;
 import com.palantir.common.exception.PalantirRuntimeException;
+import com.palantir.logsafe.SafeArg;
```
Comment (gmaretic) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java]: You're not actually using it anywhere

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java (hunk 1 above)
Comment (hsaraogi) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java]: Table names are dynamically created by certain AtlasDB clients and could contain sensitive information..

**Pull request merged**